frontend/packages/metal3-plugin/src/components/dashboard/events-card.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FirehoseResource, FirehoseResult } from '@console/internal/components/utils';
+import { EventModel, MachineModel } from '@console/internal/models';
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { EventsBody } from '@console/internal/components/dashboard/events-card/events-body';
+import { EventKind, K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { DashboardItemProps } from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { getName, getNamespace } from '@console/shared';
+
+import { getHostMachineName } from '../../selectors';
+import { BaremetalHostModel } from '../../models';
+
+const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
+const getMachineResource = (name: string, namespace: string): FirehoseResource => ({
+  isList: false,
+  namespace,
+  name,
+  kind: referenceForModel(MachineModel),
+  prop: 'machine',
+});
+
+const matchesHost = (machine: K8sResourceKind, event: EventKind): boolean =>
+  machine && _.get(event, 'source.host') === _.get(machine, 'status.nodeRef.name');
+
+const machesInvolvedObject = (host: K8sResourceKind, event: EventKind): boolean =>
+  _.isMatch(event.involvedObject, {
+    kind: referenceForModel(BaremetalHostModel),
+    name: getName(host),
+    namespace: getNamespace(host),
+  });
+
+const hostEventsFilter = (
+  host: K8sResourceKind,
+  machine: K8sResourceKind,
+  event: EventKind,
+): boolean => matchesHost(machine, event) || machesInvolvedObject(host, event);
```

Comment:
@honza @jtomasek can you check if this filtering is what you want ?

Reply:
I am currently seeing various events (from Pods, Replica Sets...) I'd expect events from BareMetalHost, Machine and Node

Reply:
When I test this on a host which does not have any machine yet, the resources.machine is a random machine from the machines list